packages/snap/src/Snap.service.ts

```diff
@@ -548,15 +550,30 @@ class SnapService {
         res = await GeneralService.togglePopups();
         return ResultObject.success(res);
       case 'addTrustedDapp':
-        if (origin === 'masca.io')
-          trustedOrigin = new URL(params.origin).hostname;
-        await GeneralService.addTrustedDapp({ origin: trustedOrigin });
+        // If the origin is masca.io, any HOSTNAME can be added. Expect parameter to be a hostname!
+        if (origin === 'masca.io') trustedOrigin = params.origin;
+        await GeneralService.addTrustedDapp({ originHostname: trustedOrigin });
         return ResultObject.success(true);
       case 'removeTrustedDapp':
-        if (origin !== 'masca.io' && origin !== new URL(params.origin).hostname)
+        if (origin !== 'masca.io' && origin !== params.origin)
           throw new Error('Unauthorized to remove other dApps');
-        await GeneralService.removeTrustedDapp({ origin: trustedOrigin });
-        return ResultObject.success(true);
+        await GeneralService.removeTrustedDapp({
+          originHostname: trustedOrigin,
+        });
+        return ResultObject.success(false);
```

Comment:
isSuccess indicates that the call was successful. Return value indicates whether or not dapp is trusted after the call of the RPC method

Reply:
If `removeTrustedDapp` is successful, we already know the dapp is not trusted, so its better to return `true` (meaning the methods was successful).

I also think we are returning booleans in those cases, because we cant return `void` (as snaps json stringify the response) and we get an error.